test/tests/functional/pbs_apbasil_inventory.py

```diff
@@ -233,31 +233,22 @@ def test_cray_login_node_memory_unit(self):
         mom_id = self.mom.shortname
         try:
             cray_login_node = self.server.status(NODE, id=mom_id)[0]
-        except PbsStatusError:
-            self.assertFalse(
-                "Cray login node %s doesn't exist on pbs server" % (mom_id))
-
-        self.mom.unset_mom_config('$alps_client', False)
-        self.reset_nodes(mom_id)
-
-        try:
+            self.mom.unset_mom_config('$alps_client', False)
+            self.reset_nodes(mom_id)
             pbs_node = self.server.status(NODE, id=mom_id)[0]
 
         except PbsStatusError:
-            self.assertFalse(
-                "Cray login node %s doesn't exist on pbs server" % (mom_id))
+            self.assertFalse(True,
+                             "Mom node %s doesn't exist on pbs server"
+                             % (mom_id))
 
         for rsc, val in pbs_node:
             if rsc in cray_login_node:
```

Comment:
Instead of this if case, do a self.assertTrue(rsc in cray_login_node, msg).  If cases where the else case is an assertFalse(True) aren't the right approach.

Reply:
Thanks for this comment, that does make sense and looks way better.